filters/auth/encrypter.go

```diff
@@ -0,0 +1,167 @@
+package auth
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	crand "crypto/rand"
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/crypto/scrypt"
+	"io"
+	"io/ioutil"
+	"strings"
+	"sync"
+	"time"
+)
+
+//secretSource operates on the secret for OpenID
+type secretSource interface {
+	getSecret() ([][]byte, error)
+}
+
+type fileSecretSource struct {
+	fileName string
+}
+
+func (fss *fileSecretSource) getSecret() ([][]byte, error) {
+	contents, err := ioutil.ReadFile(fss.fileName)
+	if err != nil {
+		return nil, err
+	}
+	secrets := strings.Split(string(contents), ",")
+	byteSecrets := make([][]byte, len(secrets))
+	for i, s := range secrets {
+		byteSecrets[i] = []byte(s)
+		if len(byteSecrets[i]) == 0 {
+			return nil, fmt.Errorf("file %s secret %d is empty", fss.fileName, i)
+		}
+	}
+	if len(byteSecrets) == 0 {
+		return nil, fmt.Errorf("secrets file %s is empty", fss.fileName)
+	}
+
+	return byteSecrets, nil
+}
+
+func newFileSecretSource(file string) secretSource {
+	return &fileSecretSource{fileName: file}
+}
+
+type encrypter struct {
+	cipherSuites []cipher.AEAD
+	mux          sync.RWMutex
+	sSource      secretSource
+	closer       chan int
```

Comment:
normally we call it `quit` and use `chan struct{}` to have a zero allocation "ping" channel